Failing tests

A bunch of tests seem to fail.
```
..FF.F.FF...F..
======================================================================
FAIL: test_alt_added_to_img (tests.test_fragments.TestFrags1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 41, in test_alt_added_to_img
    self.assertEqual(doc, expected)
AssertionError: u"<img src='foo'>" != u'<img src=\'foo\' alt="">'
- <img src='foo'>
+ <img src='foo' alt="">
?               +++++++


======================================================================
FAIL: test_entity_preserved_using_bytes (tests.test_fragments.TestFrags1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 47, in test_entity_preserved_using_bytes
    self.assertEqual(doc, expected)
AssertionError: '\xc3\xa9' != '&eacute;'

======================================================================
FAIL: test_numeric_entities_using_bytes (tests.test_fragments.TestFrags1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 53, in test_numeric_entities_using_bytes
    self.assertEqual(doc, expected)
AssertionError: '\xc3\xa9' != '&#233;'

======================================================================
FAIL: test_alt_added_to_img (tests.test_docs.TestDocs1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 53, in test_alt_added_to_img
    self.assertEqual(doc, expected)
AssertionError: u'<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <tit [truncated]... != u'<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <tit [truncated]...
  <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
  <html>
    <head>
      <title></title>
    </head>
    <body>
-     <img src='foo'>
+     <img src='foo' alt="">
?                   +++++++
    </body>
  </html>


======================================================================
FAIL: test_entity_preserved_using_bytes (tests.test_docs.TestDocs1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 59, in test_entity_preserved_using_bytes
    self.assertEqual(doc, expected)
AssertionError: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    \xc3\xa9\n  </body>\n</html>\n' != '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    &eacute;\n  </body>\n</html>\n'

======================================================================
FAIL: test_numeric_entities_using_bytes (tests.test_docs.TestDocs1)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 65, in test_numeric_entities_using_bytes
    self.assertEqual(doc, expected)
AssertionError: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    \xc3\xa9\n  </body>\n</html>\n' != '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    &#233;\n  </body>\n</html>\n'

----------------------------------------------------------------------
Ran 15 tests in 0.078s

FAILED (failures=6)
```

What version of tidy library is this with?

> On Jan 24, 2016, at 2:53 AM, Frederik Rietdijk <notifications@github.com> wrote:
> 
> A bunch of tests seem to fail.
> 
> ..FF.F.FF...F..
> ======================================================================
> FAIL: test_alt_added_to_img (tests.test_fragments.TestFrags1)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 41, in test_alt_added_to_img
>     self.assertEqual(doc, expected)
> AssertionError: u"<img src='foo'>" != u'<img src=\'foo\' alt="">'
> - <img src='foo'>
> + <img src='foo' alt="">
> ?               +++++++
> 
> 
> ======================================================================
> FAIL: test_entity_preserved_using_bytes (tests.test_fragments.TestFrags1)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 47, in test_entity_preserved_using_bytes
>     self.assertEqual(doc, expected)
> AssertionError: '\xc3\xa9' != '&eacute;'
> 
> ======================================================================
> FAIL: test_numeric_entities_using_bytes (tests.test_fragments.TestFrags1)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_fragments.py", line 53, in test_numeric_entities_using_bytes
>     self.assertEqual(doc, expected)
> AssertionError: '\xc3\xa9' != '&#233;'
> 
> ======================================================================
> FAIL: test_alt_added_to_img (tests.test_docs.TestDocs1)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 53, in test_alt_added_to_img
>     self.assertEqual(doc, expected)
> AssertionError: u'<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <tit [truncated]... != u'<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <tit [truncated]...
>   <!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
>   <html>
>     <head>
>       <title></title>
>     </head>
>     <body>
> -     <img src='foo'>
> +     <img src='foo' alt="">
> ?                   +++++++
>     </body>
>   </html>
> 
> 
> ======================================================================
> FAIL: test_entity_preserved_using_bytes (tests.test_docs.TestDocs1)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 59, in test_entity_preserved_using_bytes
>     self.assertEqual(doc, expected)
> AssertionError: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    \xc3\xa9\n  </body>\n</html>\n' != '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    &eacute;\n  </body>\n</html>\n'
> 
> ======================================================================
> FAIL: test_numeric_entities_using_bytes (tests.test_docs.TestDocs1)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/tmp/nix-build-python2.7-pytidylib-0.2.4.drv-0/pytidylib-0.2.4/tests/test_docs.py", line 65, in test_numeric_entities_using_bytes
>     self.assertEqual(doc, expected)
> AssertionError: '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    \xc3\xa9\n  </body>\n</html>\n' != '<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">\n<html>\n  <head>\n    <title></title>\n  </head>\n  <body>\n    &#233;\n  </body>\n</html>\n'
> 
> ----------------------------------------------------------------------
> Ran 15 tests in 0.078s
> 
> FAILED (failures=6)
> —
> Reply to this email directly or view it on GitHub <https://github.com/countergram/pytidylib/issues/13>.
>

--- a/tests/test_docs.py
+++ b/tests/test_docs.py
@@ -48,20 +48,20 @@ class TestDocs1(unittest.TestCase):
 
     def test_alt_added_to_img(self):
         h = "<img src='foo'>"
-        expected = DOC % '''<img src='foo' alt="">'''
-        doc, err = tidy_document(h)
+        expected = DOC % '''<img src='foo' alt="bar">'''
+        doc, err = tidy_document(h, {'alt-text': 'bar'})
         self.assertEqual(doc, expected)
 
     def test_entity_preserved_using_bytes(self):
         h = b"&eacute;"
         expected = (DOC % "&eacute;").encode('utf-8')
-        doc, err = tidy_document(h)
+        doc, err = tidy_document(h, {'preserve-entities': 1})
         self.assertEqual(doc, expected)
 
     def test_numeric_entities_using_bytes(self):
         h = b"&eacute;"
         expected = (DOC % "&#233;").encode('utf-8')
-        doc, err = tidy_document(h, {'numeric-entities': 1})
+        doc, err = tidy_document(h, {'numeric-entities': 1, 'output-encoding': 'ascii'})
         self.assertEqual(doc, expected)
 
     def test_non_ascii_preserved(self):

--- a/tests/test_fragments.py
+++ b/tests/test_fragments.py
@@ -36,20 +36,20 @@ class TestFrags1(unittest.TestCase):
 
     def test_alt_added_to_img(self):
         h = "<img src='foo'>"
-        expected = '''<img src='foo' alt="">'''
-        doc, err = tidy_fragment(h)
+        expected = '''<img src='foo' alt="bar">'''
+        doc, err = tidy_fragment(h, {'alt-text': 'bar'})
         self.assertEqual(doc, expected)
 
     def test_entity_preserved_using_bytes(self):
         h = b"&eacute;"
         expected = b"&eacute;"
-        doc, err = tidy_fragment(h)
+        doc, err = tidy_fragment(h, {'preserve-entities': 1})
         self.assertEqual(doc, expected)
 
     def test_numeric_entities_using_bytes(self):
         h = b"&eacute;"
         expected = b"&#233;"
-        doc, err = tidy_fragment(h, {'numeric-entities': 1})
+        doc, err = tidy_fragment(h, {'numeric-entities': 1, 'output-encoding': 'ascii'})
         self.assertEqual(doc, expected)
 
     def test_non_ascii_preserved(self):

